client/app/cloud/project/billing/cloud-project-billing-legacy.service.js

```diff
@@ -0,0 +1,297 @@
+        return getEmptyConsumption('hourlyInstances');
+      }
+      const hourlyInstances = _.flatten(_.map(
+        _.get(consumptionDetails, 'hourlyBilling.hourlyUsage.instance'),
+        instance => _.map(instance.details, (detail) => {
+          const newDetail = _.clone(detail);
+          newDetail.totalPrice = roundNumber(newDetail.totalPrice, 2);
+          return _.extend(newDetail, { reference: instance.reference, region: instance.region });
+        }),
+      ));
+      const hourlyInstancesTotal = _.reduce(
+        _.get(consumptionDetails, 'hourlyBilling.hourlyUsage.instance'),
+        (sum, instance) => sum + roundNumber(instance.totalPrice, 2), 0,
+      );
+
+      return ({ hourlyInstances, total: roundNumber(hourlyInstancesTotal, 2) });
+    }
+
+    function initMonthlyInstanceList(consumptionDetails) {
+      if (!_.get(consumptionDetails, 'monthlyBilling') || !_.get(consumptionDetails, 'monthlyBilling.monthlyUsage')) {
+        return getEmptyConsumption('monthlyInstances');
+      }
+
+      const monthlyInstances = _.flatten(_.map(
+        _.get(consumptionDetails, 'monthlyBilling.monthlyUsage.instance'),
+        instance => _.map(instance.details, (detail) => {
+          const newDetail = _.clone(detail);
+          newDetail.totalPrice = roundNumber(newDetail.totalPrice, 2);
+          return _.extend(newDetail, { reference: instance.reference, region: instance.region });
+        }),
+      ));
+
+      const monthlyTotal = _.reduce(
+        _.get(consumptionDetails, 'monthlyBilling.monthlyUsage.instance'),
+        (sum, instance) => sum + roundNumber(instance.totalPrice, 2),
+        0,
+      );
+
+      return ({
+        monthlyInstances,
+        total: roundNumber(monthlyTotal, 2),
+      });
+    }
+
+    function initObjectStorageList(consumptionDetails) {
+      if (!consumptionDetails.hourlyBilling || !consumptionDetails.hourlyBilling.hourlyUsage) {
+        return getEmptyConsumption('objectStorages');
+      }
+
+      const objectStorages = _.reject(
+        _.map(
+          consumptionDetails.hourlyBilling.hourlyUsage.storage,
+          objectStorage => _.set(objectStorage, 'totalPrice', roundNumber(objectStorage.totalPrice, 2)),
+        ), { type: 'pca' },
+      );
+
+      const objectStorageTotal = _.reduce(
+        objectStorages,
+        (sum, storage) => sum + roundNumber(storage.totalPrice, 2),
+        0,
+      );
+
+      return ({
+        objectStorages,
+        total: roundNumber(objectStorageTotal, 2),
+      });
+    }
+
+    function initArchiveStorageList(consumptionDetails) {
+      if (!consumptionDetails.hourlyBilling || !consumptionDetails.hourlyBilling.hourlyUsage) {
+        return getEmptyConsumption('archiveStorages');
+      }
+
+      const archiveStorages = _.filter(
+        _.map(
+          consumptionDetails.hourlyBilling.hourlyUsage.storage,
+          objectStorage => _.set(objectStorage, 'totalPrice', roundNumber(objectStorage.totalPrice, 2)),
```

Comment:
Wouldn't you prefer to create a new object instead of changing the argument ? :)

```js
objectStorage => ({
    ...objectStorage,
    totalPrice: roundNumber(objectStorage.totalPrice, 2),
})
```